src/amo/reducers/collections.js

```diff
@@ -898,16 +898,14 @@ export const loadCollectionIntoState = ({
   pageSize,
 }: LoadCollectionIntoStateParams): CollectionsState => {
   const existingCollection = state.byId[collection.id];
-  const internalCollection = createInternalCollection({
-    detail: collection,
-    items: addons,
-    pageSize,
-  });
-  // In case the new collection isn't loaded with add-ons,
-  // make sure we don't overwrite any existing addons.
-  if (!internalCollection.addons && existingCollection) {
-    internalCollection.addons = existingCollection.addons;
-  }
+  const internalCollection =
```

Comment:
This works, but it always loads everything from the `existingCollection` if one exists and there are no `addons`, so it will ignore anything in the `collection` parameter, which may not be what we want.

It might be better to add a line into the `if` block at line 908 to also take `pageSize` from `existingCollection` and use it in `internalCollection` just as we've done with `addons`:

```diff
diff --git a/src/amo/reducers/collections.js b/src/amo/reducers/collections.js
index 5ee5cee..745a4ee 100644
--- a/src/amo/reducers/collections.js
+++ b/src/amo/reducers/collections.js
@@ -898,14 +898,17 @@ export const loadCollectionIntoState = ({
   pageSize,
 }: LoadCollectionIntoStateParams): CollectionsState => {
   const existingCollection = state.byId[collection.id];
-  const internalCollection =
-    !addons && existingCollection
-      ? existingCollection
-      : createInternalCollection({
-          detail: collection,
-          items: addons,
-          pageSize,
-        });
+  const internalCollection = createInternalCollection({
+    detail: collection,
+    items: addons,
+    pageSize,
+  });
+  // In case the new collection isn't loaded with add-ons,
+  // make sure we don't overwrite any existing addons.
+  if (!internalCollection.addons && existingCollection) {
+    internalCollection.addons = existingCollection.addons;
+    internalCollection.pageSize = existingCollection.pageSize;
+  }

   return {
     ...state,
```

Also, the test for the reducer should be updated to include a test for this new use case.

Reply:
Ok, I see.
Sorry, my computer was broken 😭
I will fix it soon.

Reply:
@bobsilverberg 
Thanks for your advise.
I have updated it. 
Please check.